extras/indexing/src/main/java/org/apache/rya/indexing/entity/storage/CloseableIterator.java

```diff
@@ -0,0 +1,29 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.rya.indexing.entity.storage;
+
+import java.io.Closeable;
+import java.util.Iterator;
+
+/**
+ * An {@link Iterator} that is also {@link Closeable}.
+ *
+ * @param <T> - The type of object that will be iterated over.
+ */
+public interface CloseableIterator<T> extends Iterator<T>, Closeable { }
```

Comment:
i'm not good with merging this until this is either removed or renamed to not be the same name as the mango implementation.  it reminds me of places where you see custom lists (like arraylist), which i would like to avoid introducing if we can.
If you won't refactor to use the mango implementation, then rename to something other than CloseableIterator.  Once that's done I think this is good to merge.